test_requests.py

```diff
@@ -188,6 +188,28 @@ def test_HTTP_302_ALLOW_REDIRECT_GET(self, httpbin):
         assert r.history[0].status_code == 302
         assert r.history[0].is_redirect
 
+    def test_HTTP_302_TOO_MANY_REDIRECTS(self, httpbin):
+        try:
+            requests.get(httpbin('redirect', '50'))
+        except TooManyRedirects as e:
+            assert '/relative-redirect/20' in e.request.url
+            assert e.request.url == e.response.url
+            assert len(e.response.history) == 30
+        else:
+            pytest.fail()
+
+    def test_HTTP_302_TOO_MANY_REDIRECTS_WITH_PARAMS(self, httpbin):
+        s = requests.session()
+        s.max_redirects = 5
+        try:
+            s.get(httpbin('redirect', '50'))
+        except TooManyRedirects as e:
+            assert '/relative-redirect/45' in e.request.url
+            assert e.request.url == e.response.url
+            assert len(e.response.history) == 5
+        else:
+            pytest.fail()
```

Comment:
This message might be slightly different, e.g., "Expected custom max number of redirects to be respected but was not."